Modules/Registration/Common/include/itkPointsLocator.hxx

```diff
@@ -101,6 +119,24 @@ PointsLocator<TPointsContainer>::FindClosestNPoints(const PointType &         qu
   this->m_Tree->Search(query, N, identifiers);
 }
 
+template <typename TPointsContainer>
+void
+PointsLocator<TPointsContainer>::FindClosestNPoints(const PointType &         query,
```

Comment:
Why not delegate this to `Search` or `Search` to this? Currently, they are 100% duplicated.

Reply:
Yes, I think this interface was designed before and it is still present for compatibility.
Can be removed. I will check more in the codebase why two separate methods are present here.

Reply:
The Search method is present to keep the nomenclature of underlying KDtree and PointsLocator the same.

Personally, I want to keep the **FindClosestNPoints** and **FindPointsWithinRadius** because they are 
easy to find for someone.

If we want to remove then I would pick the **Search** method.

Comments are welcome here.

Reply:
I am fine with either. As you are doing the work, follow your preference.